Start collection of AWX examples

jhernand opened the pull request:
This pull requests starts a collection of AWX examples. They will be initially
in the `examples/awx` directory.

In order to make examples simpler the pull request also adds a new `CAFile` method to the AWX connection builder, as that simplifies loading CA certificates.

jhernand commented:
@elad661 @moolitayer @zgalor @nimrodshn please review, and consider contributing your own examples.

moolitayer commented on examples/awx/README.md:
```diff
@@ -0,0 +1,25 @@
+# AWX Client Examples
```
Consider adding a link to this file from the root README for visibility.

jhernand replied:
Done in the next version of the pull request.

moolitayer commented on examples/awx/list_job_templates.go:
```diff
@@ -0,0 +1,90 @@
+//		-url "https://awx.example.com/api" \
+//		-user "admin" \
+//		-password "..." \
+//		-ca "ca.pem" \
```
Could you name this field & and function caFile? it was confusing for me to read `-ca "ca.pem"`. Naming the parameters the same as the connections functions improves the example readability.

jhernand replied:
Done in the next version of the pull request.

moolitayer commented on pkg/awx/connection.go:
```diff
@@ -157,19 +174,59 @@ func (b *ConnectionBuilder) Build() (c *Connection, err error) {
+					}
+				}
+			}
+		}
```
The ca files addition lgtm :+1:

jhernand commented:
Updated the names of the flags so that they match the names of the methods of the builder, and added the reference from the root `README.md` of the project: https://github.com/jhernand/autoheal/tree/start_collection_of_awx_examples#dependencies

moolitayer commented on README.md:
```diff
@@ -252,9 +252,16 @@ $ autoheal server --config-file=my.yml --logtostderr
+In order to use the AWX API the auto-heal service uses a Go AWX client that is
+part of this repository, but that will be likely moved to a separate repository
+in the future. The code is in the [pkg/awx](pkg/awx) directory, and there is a
+collection of examples in the [examples/awx](examples/awx) directory.
```
:+1:

moolitayer commented:
Works for me (the new test file)
